### PR TITLE
BLD: require Meson 1.5.0, fix a missing build dependency for `cython_special`

### PR DESCRIPTION
**Pull request by rgommers**
Also add a CI check to catch this.

This was technically a race condition (no one seems to have hit it yet though). From the added CI step:

```
Run ninja -C build -t missingdeps
Missing dep: scipy/special/cython_special.cpython-310-x86_64-linux-gnu.so.p/meson-generated_cython_special.c.o uses scipy/special/_ufuncs_defs.h (generated by CUSTOM_COMMAND)
Processed 1310 nodes.
Error: There are 1 missing dependency paths.
1 targets had depfile dependencies on 1 distinct generated inputs (from 1 rules)  without a non-depfile dep path to the generator.
There might be build flakiness if any of the targets listed above are built alone, or not late enough, in a clean output directory.
```

### CODE REVIEW

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -124,6 +124,9 @@ jobs:
         python dev.py build --tags=runtime,python-runtime,devel,tests
         python tools/check_installation.py ${{ env.INSTALLDIR }}
 
+    - name: Check build-internal dependencies
+      run: ninja -C build -t missingdeps
```
Comment (tylerjereddy) [inline .github/workflows/linux.yml:128]: This is really cool (could have saved me a bisection a few weeks ago) and I checked locally that indeed it goes from exit code `3` to `0` with your patch. I think we need a `meson` version bump to get the CI passing in some places.

File: scipy/special/meson.build [hunk 2]
```diff
@@ -133,12 +133,12 @@ cython_special_pxd = fs.copyfile('cython_special.pxd')
 uf_cython_gen = generator(cython,
   arguments : cython_args,
   output : '@BASENAME@.c',
-  depends : [_cython_tree, _ufuncs_pxi_pxd_sources, cython_special_pxd])
+  depends : [_cython_tree, _ufuncs_pxi_pxd_sources, cython_special_pxd, cython_special[0]])
```
Comment (tylerjereddy) [inline scipy/special/meson.build:136]: I can reproduce the CI `meson` typing complaint locally if I downgrade to the `meson` `1.4.0` used by the Windows Intel job. Do you remember if there's a specific need for that pin? IIRC these oneAPI CI additions are new enough that it is probably fresh in someone's mind.

File: .github/workflows/linux.yml (hunk 1 above)
Comment (rgommers) [reply .github/workflows/linux.yml:128]: Yes indeed. That should be okay, and the thing that broke in CI is a quite useful change in Meson that allowed some more cleanups.

File: scipy/special/meson.build (hunk 2 above)
Comment (rgommers) [reply scipy/special/meson.build:136]: There wasn't a specific reason for this I believe.

**Comment by tylerjereddy** [comment]
All comments addressed, remaining CI failure is unrelated/seen elsewhere, and includes more cleanups now. Merged, thanks.

**Comment by rgommers** [comment]
For future reference: this will catch most, but not all, potential issues. The exceptions are things that go through manual handling (e.g., `custom_target` with codegen, like the few `_generate_pyx.py` scripts) and don't use a depfile. The reason is that there is no way for `ninja` to detect that some `.py` script internally uses `open(some_other_file).read()`.

**Pull request merged**